File: bronze/25304/c++/main.cpp
#include <iostream>

int main()
{
    int inPrice, totalPrice = 0, count, price, num;
    std::cin >> inPrice >> count;
    for (int i = 0; i < count; i++) {
        std::cin >> price >> num;
        totalPrice += (price * num);
    }
    std::cout << (inPrice == totalPrice ? "Yes" : "No");
    return 0;
}

File: bronze/25304/python/main.py

if (__name__ == "__main__"):
    inPrice = int(input())
    count = int(input())
    totalPrice = 0
    for i in range(count):
        price, num = map(int, input().split())
        totalPrice += (price * num)
    print("Yes" if totalPrice == inPrice else "No")